File: leetcode-557_reverseWordsInAStringIII.js
//给定一个字符串, 逐个单词反转
var reverseWords = function(s) {
    let arr = s.split(" "); //先切割成单词数组
    let i = 0;
    while (arr[i]) {
        arr[i] = arr[i].split("").reverse().join(""); //逐个单词反转
        i++;
    }
    return arr.join(" "); //再拼接回来
};

reverseWords("Let's take LeetCode contest"); // output: "s'teL ekat edoCteeL tsetnoc"

File: leetcode-557_reverse-words-in-a-string-iii.java
/**
 * 给定一个字符串，反转每一个单词
 */
public String reverseWords(String s) {
    String[] strs = s.split(" "); //先切出每一个单词
    for (int i = 0; i < strs.length; i++) {
        strs[i] = new StringBuffer(strs[i]).reverse().toString(); //反转单词
    }
    return String.join(" ", strs); //再拼回字符串
}
